src/shell.zig

```diff
@@ -313,3 +313,50 @@ test "shell: expand_argv" {
         &.{ "zig", "version", "--verbose" },
     );
 }
+
+const Snap = @import("./snaptest.zig").Snap;
+const snap = Snap.snap;
+
+test "shell: expand_argv snap" {
+    const T = struct {
+        fn check(
+            comptime cmd: []const u8,
+            args: anytype,
+            want: Snap,
+        ) !void {
+            var argv = Argv.init(std.testing.allocator);
+            defer argv.deinit();
+
+            try expand_argv(&argv, cmd, args);
+
+            var buf = std.ArrayList(u8).init(std.testing.allocator);
+            defer buf.deinit();
+
+            try std.json.stringify(argv.items, .{ }, buf.writer());
+            try want.diff(buf.items);
+        }
+    };
+
+    try T.check("zig version", .{}, snap(@src(),
+        \\["zig","version"]
+    ));
+    try T.check("  zig  version  ", .{}, snap(@src(),
+        \\["zig","version"]
+    ));
```

Comment:
Here's how robustness to refactors plays out. Suppose that I want to extend the syntax of `expand_argv` to support not only the command line itself, but also environment variables: 

```zig
expand_argv("LD_PRELOAD={faulty_libc} tigerbeetle format", .{ faulty_libc = "..." })
```

This requires me to change the return type from `[]const []const u8` to something like `struct { env: EnvMap, args: []const []const u8}`. The `snap` thing would allow me to re-run the existing test suit with `SNAP_UPDATE=1` to fix all existing tests, without actually touching the code.